### PR TITLE
Query improvements javadoc fixes

### PR DESCRIPTION
**Pull request by tomblench**
## What

Javadoc fixes and clarifications for query improvements branch: better examples and clearer guidance for users browsing documents.

## Testing

`./gradlew javadoc` runs OK.

### CODE REVIEW

File: cloudant-client/src/main/java/com/cloudant/client/api/query/package-info.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * </p>
+ * <h1>Overview</h1>
+ * <p>
+ * This package contains classes which make it easier to interact with the query API in the
+ * following ways:
+ * </p>
+ * <ul>
+ *     <li>
+ *         {@link com.cloudant.client.api.query.QueryBuilder A class for building complex queries}
+ *         without having to write them "manually" in JSON.
```
Comment (ricellis) [inline cloudant-client/src/main/java/com/cloudant/client/api/query/package-info.java]: `write them "manually" in JSON`
I know what you mean and the use of `"` suggests to me that you were trying to think of a better way to phrase this, fwiw the best I can think of is
>without having to compose the string representation of the query JSON

which isn't a shining exemplar of the English language :)

File: cloudant-client/src/main/java/com/cloudant/client/api/Database.java [hunk 2]
```diff
@@ -328,13 +328,32 @@ public void createIndex(String indexName, String designDocName, String indexType
      * </P>
      * <pre>
      * {@code
-     * // Create a JSON index with generated names for the field named "Movie_year" with the default
-     * // ascending sort order
-     * db.createIndex(new JsonIndex.Builder().fields(new JsonIndex.Field("Movie_year")).definition());
+     * // Create a JSON index with a generated name for the field named "Movie_year" with ascending
```
[the remark this answers is not in the document]
Comment (tomblench) [reply cloudant-client/src/main/java/com/cloudant/client/api/Database.java]: See 00dd1a0 which is more consistent with the style used in `query` javadoc

File: cloudant-client/src/main/java/com/cloudant/client/api/Database.java [hunk 3]
```diff
@@ -328,13 +328,32 @@ public void createIndex(String indexName, String designDocName, String indexType
...
+     *     desc("Movie_year").
+     *     partialFilterSelector(selector).
+     *     definition();
+     * db.createIndex(indexDefinition);
+     *
+     * // Create a text index with a generated name for the string field named "Movie_title"
+     * db.createIndex(TextIndex.builder().string("Movie_title").definition());
+     *
+     * // Create a partial text index named "movies-after-2010" for the string field named
+     * // "Movie_title" which will index all movies with "Movie_year" greater than 2010
```
[the remark this answers is not in the document]
Comment (tomblench) [reply cloudant-client/src/main/java/com/cloudant/client/api/Database.java]: Done in 00dd1a0

File: cloudant-client/src/main/java/com/cloudant/client/api/Database.java [hunk 4]
```diff
@@ -328,13 +328,32 @@ public void createIndex(String indexName, String designDocName, String indexType
...
+     * Selector selector = gt("Movie_year", 2010);
+     * String indexDefinition = TextIndex.builder().
+     *     string("Movie_title").
+     *     name("movies-after-2010").
```
[the remark this answers is not in the document]
Comment (tomblench) [reply cloudant-client/src/main/java/com/cloudant/client/api/Database.java]: Done in 00dd1a0

File: cloudant-client/src/main/java/com/cloudant/client/api/query/package-info.java (hunk 1 above)
Comment (tomblench) [reply cloudant-client/src/main/java/com/cloudant/client/api/query/package-info.java]: Yes I was struggling for the right expression - how does 034f6ec look?

File: cloudant-client/src/main/java/com/cloudant/client/api/query/package-info.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2018 IBM Corp. All rights reserved.
```
[the remark this answers is not in the document]
Comment (tomblench) [reply cloudant-client/src/main/java/com/cloudant/client/api/query/package-info.java]: Done in 034f6ec

File: cloudant-client/src/main/java/com/cloudant/client/api/query/TextIndex.java [hunk 6]
```diff
@@ -200,11 +244,35 @@ protected TextIndex newInstance() {
...
-         * // TODO string, object differences and example
+         * The analyzer can be a simple string or a more complex JSON object, as shown in the
+         * following examples:
+         * </P>
+         * <P>
+         * Configure the "keyword" analyzer:
          * </P>
+         * <pre>
+         * {@code
+         * TextIndex.builder().analyzer("keyword);
```
Comment (emlaver) [inline cloudant-client/src/main/java/com/cloudant/client/api/query/TextIndex.java]: Missing parentheses around `keyword`.

**Pull request merged**